### PR TITLE
Add dependabot

### PR DESCRIPTION
**Pull request by egecetin**

### CODE REVIEW

**Comment by clementperon** [comment]
Is it possible to force a manual run, just to check what is the exepected result of this bot ?

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+version: 2
+updates:
```
Comment (seladb) [inline .github/dependabot.yml:2]: We can add another update for `pip`: https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#pip-and-pip-compile

To update the ExamplesTest and our CI script, as this PR is doing: https://github.com/seladb/PcapPlusPlus/pull/1283

cc @clementperon

**Review by seladb — COMMENTED** [review]
@egecetin Maybe we should rename the PR title to "Add dependabot"?

**Comment by egecetin** [comment]
@clementperon I'll check it. Even if there is no way since dependabot only opens PRs we can just close PRs without merging them.

@seladb It's ok for me.

Also i noticed something I forgot since I don't target any branches probably dependabot will target master branch. I'll fix it.

**Review by seladb — APPROVED** [review]

**Comment by egecetin** [comment]
@clementperon I found a way to dry-run locally using dependabot-core but not tested yet so I'm not sure it gives the results we desired. I'll ping you when i have the results.

**Comment by egecetin** [comment]
@clementperon @seladb For github-actions everything is ok. `Dependabot CLI` generates following result,

```
updater | 2024/01/23 07:46:13 INFO Results:
updater | +-----------------------------------------------------------+
updater | |            Changes to Dependabot Pull Requests            |
updater | +---------+-------------------------------------------------+
updater | | created | actions/checkout ( from 3 to 4 )                |
updater | | created | actions/setup-python ( from 4 to 5 )            |
updater | | created | peter-evans/create-pull-request ( from 4 to 5 ) |
updater | | created | actions/upload-artifact ( from 1 to 4 )         |
updater | | created | github/codeql-action ( from 2 to 3 )            |
updater | | created | vmactions/freebsd-vm ( from 0 to 1 )            |
updater | | created | actions/download-artifact ( from 3 to 4 )       |
updater | +---------+-------------------------------------------------+
  proxy | 2024/01/23 07:46:13 58/115 calls cached (50%)
```

but for `pip` there is a problem. It thinks `codespell-ignore-list.txt` as a requirements.txt and tries to update it and not searches the real requirements file in Tests directory

```
updater | 2024/01/23 07:49:54 INFO Starting update job for seladb/PcapPlusPlus
updater | 2024/01/23 07:49:54 INFO Checking all dependencies for version updates...
updater | 2024/01/23 07:49:54 INFO Checking if hist  needs updating
  proxy | 2024/01/23 07:49:54 [010] GET https://pypi.org:443/simple/hist/
  proxy | 2024/01/23 07:49:54 [010] 200 https://pypi.org:443/simple/hist/
updater | 2024/01/23 07:49:54 INFO Latest version is 2.7.2
  proxy | 2024/01/23 07:49:54 [012] GET https://pypi.org:443/simple/hist/
  proxy | 2024/01/23 07:49:54 [012] 200 https://pypi.org:443/simple/hist/
updater | 2024/01/23 07:49:54 INFO No update needed for hist 
updater | 2024/01/23 07:49:54 INFO Checking if filetests  needs updating
  proxy | 2024/01/23 07:49:55 [014] GET https://pypi.org:443/simple/filetests/
  proxy | 2024/01/23 07:49:55 [014] 404 https://pypi.org:443/simple/filetests/
updater | 2024/01/23 07:49:55 INFO Latest version is 
  proxy | 2024/01/23 07:49:55 [016] GET https://pypi.org:443/simple/filetests/
  proxy | 2024/01/23 07:49:55 [016] 404 https://pypi.org:443/simple/filetests/
updater | 2024/01/23 07:49:55 INFO No update needed for filetests 

```

**Comment by seladb** [comment]
> @clementperon @seladb For github-actions everything is ok. `Dependabot CLI` generates following result,
> 
> ```
> updater | 2024/01/23 07:46:13 INFO Results:
> updater | +-----------------------------------------------------------+
> updater | |            Changes to Dependabot Pull Requests            |
> updater | +---------+-------------------------------------------------+
> updater | | created | actions/checkout ( from 3 to 4 )                |
> updater | | created | actions/setup-python ( from 4 to 5 )            |
> updater | | created | peter-evans/create-pull-request ( from 4 to 5 ) |
> updater | | created | actions/upload-artifact ( from 1 to 4 )         |
> updater | | created | github/codeql-action ( from 2 to 3 )            |
> updater | | created | vmactions/freebsd-vm ( from 0 to 1 )            |
> updater | | created | actions/download-artifact ( from 3 to 4 )       |
> updater | +---------+-------------------------------------------------+
>   proxy | 2024/01/23 07:46:13 58/115 calls cached (50%)
> ```
> 
> but for `pip` there is a problem. It thinks `codespell-ignore-list.txt` as a requirements.txt and tries to update it and not searches the real requirements file in Tests directory
> 
> ```
> updater | 2024/01/23 07:49:54 INFO Starting update job for seladb/PcapPlusPlus
> updater | 2024/01/23 07:49:54 INFO Checking all dependencies for version updates...
> updater | 2024/01/23 07:49:54 INFO Checking if hist  needs updating
>   proxy | 2024/01/23 07:49:54 [010] GET https://pypi.org:443/simple/hist/
>   proxy | 2024/01/23 07:49:54 [010] 200 https://pypi.org:443/simple/hist/
> updater | 2024/01/23 07:49:54 INFO Latest version is 2.7.2
>   proxy | 2024/01/23 07:49:54 [012] GET https://pypi.org:443/simple/hist/
>   proxy | 2024/01/23 07:49:54 [012] 200 https://pypi.org:443/simple/hist/
> updater | 2024/01/23 07:49:54 INFO No update needed for hist 
> updater | 2024/01/23 07:49:54 INFO Checking if filetests  needs updating
>   proxy | 2024/01/23 07:49:55 [014] GET https://pypi.org:443/simple/filetests/
>   proxy | 2024/01/23 07:49:55 [014] 404 https://pypi.org:443/simple/filetests/
> updater | 2024/01/23 07:49:55 INFO Latest version is 
>   proxy | 2024/01/23 07:49:55 [016] GET https://pypi.org:443/simple/filetests/
>   proxy | 2024/01/23 07:49:55 [016] 404 https://pypi.org:443/simple/filetests/
> updater | 2024/01/23 07:49:55 INFO No update needed for filetests 
> ```

Maybe we should add 2 different `"pip"` sections, one for the directory `ci/run_tests` and one for the directory `Test/ExamplesTest`?

**Comment by egecetin** [comment]
Yes it fixed when I specified the directory for example for `Tests/ExampleTest`

```
updater | 2024/01/23 08:08:05 INFO Results:
updater | +---------------------------------------------+
updater | |     Changes to Dependabot Pull Requests     |
updater | +---------+-----------------------------------+
updater | | created | attrs ( from 21.4.0 to 23.2.0 )   |
updater | | created | iniconfig ( from 1.1.1 to 2.0.0 ) |
updater | | created | pyparsing ( from 3.0.7 to 3.1.1 ) |
updater | | created | pytest ( from 6.2.5 to 7.4.4 )    |
updater | | created | scapy ( from 2.4.5 to 2.5.0 )     |
updater | +---------+-----------------------------------+
  proxy | 2024/01/23 08:08:05 39/61 calls cached (63%)

```

And not related but FreeBSD gives error

**Comment by seladb** [comment]
> Yes it fixed when I specified the directory for example for `Tests/ExampleTest`
> 
> ```
> updater | 2024/01/23 08:08:05 INFO Results:
> updater | +---------------------------------------------+
> updater | |     Changes to Dependabot Pull Requests     |
> updater | +---------+-----------------------------------+
> updater | | created | attrs ( from 21.4.0 to 23.2.0 )   |
> updater | | created | iniconfig ( from 1.1.1 to 2.0.0 ) |
> updater | | created | pyparsing ( from 3.0.7 to 3.1.1 ) |
> updater | | created | pytest ( from 6.2.5 to 7.4.4 )    |
> updater | | created | scapy ( from 2.4.5 to 2.5.0 )     |
> updater | +---------+-----------------------------------+
>   proxy | 2024/01/23 08:08:05 39/61 calls cached (63%)
> ```
> 
> And not related but FreeBSD gives error

This PR update FreeBSD to 13.2: https://github.com/seladb/PcapPlusPlus/pull/1281

**Review by clementperon — APPROVED** [review]

**Pull request merged**